app/models/taxonomy.rb

```diff
@@ -37,9 +37,9 @@ def self.inherited(child)
       scoped_search :on => :description, :complete_enabled => :false, :only_explicit => true
       scoped_search :on => :id, :validator => ScopedSearch::Validators::INTEGER
 
-      apipie :class, desc: "A class representing #{model_name.human} object" do
+      apipie :class, desc: "A class representing #{model_name.singular.humanize(capitalize: false)} object" do
         sections only: %w[all additional]
-        name_exl, title_exl = class_scope.model_name.human == 'Location' ? ['Europe', 'Europe/Prague'] : ['Red Hat', 'Red Hat/Engineering']
+        name_exl, title_exl = model_name.to_s == 'Location' ? ['Europe', 'Europe/Prague'] : ['Red Hat', 'Red Hat/Engineering']
```

Comment:
@ekohl I believe we can consider using `class_scope` here to determine the context of the model and this will help us to resolve the `apipie` issue.

``` ruby

name_exl, title_exl = class_scope.model_name.to_s == 'Location' ? ['Europe', 'Europe/Prague'] : ['Red Hat', 'Red Hat/Engineering']


```

Reply:
Oh, good catch.